File: it/test-versions/expected/suite/java/test/Entry.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import foo._4_0_0.Thing;
import java.util.Objects;
import java.util.Optional;

public class Entry {
  @JsonProperty("thing")
  private final Optional<Thing> thing;

  @JsonCreator
  public Entry(
    @JsonProperty("thing") final Optional<Thing> thing
  ) {
    Objects.requireNonNull(thing, "thing");
    this.thing = thing;
  }

  public Optional<Thing> getThing() {
    return this.thing;
  }

  @Override
  public int hashCode() {
    int result = 1;
    result = result * 31 + this.thing.hashCode();
    return result;
  }

  @Override
  public boolean equals(final Object other) {
    if (other == null) {
      return false;
    }

    if (!(other instanceof Entry)) {
      return false;
    }

    @SuppressWarnings("unchecked")
    final Entry o = (Entry) other;

    if (!this.thing.equals(o.thing)) {
      return false;
    }

    return true;
  }

  @Override
  public String toString() {
    final StringBuilder b = new StringBuilder();

    b.append("Entry");
    b.append("(");
    b.append("thing=");
    b.append(this.thing.toString());
    b.append(")");

    return b.toString();
  }

  public static class Builder {
    private Optional<Thing> thing = Optional.empty();

    public Builder thing(final Thing thing) {
      this.thing = Optional.of(thing);
      return this;
    }

    public Entry build() {
      final Optional<Thing> thing = this.thing;

      return new Entry(thing);
    }
  }
}

File: it/test-versions/expected/suite/csharp/Test/Entry.cs
using System;

namespace Test {
  [Newtonsoft.Json.JsonObject(ItemNullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
  public class Entry {
    [Newtonsoft.Json.JsonProperty("thing")]
    public Foo._4_0_0.Thing thing {
      get;
    }

    [Newtonsoft.Json.JsonConstructor]
    public Entry(
      [Newtonsoft.Json.JsonProperty("thing")] Foo._4_0_0.Thing thing
    ) {
      this.thing = thing;
    }

    public override Int32 GetHashCode() {
      Int32 result = 1;
      result = result * 31 + this.thing.GetHashCode();
      return result;
    }

    public override Boolean Equals(System.Object other) {
      Entry o = other as Entry;

      if (o == null) {
        return false;
      }

      if (this.thing == null) {
        if (o.thing != null) {
          return false;
        }
      } else {
        if (!this.thing.Equals(o.thing)) {
          return false;
        }
      }

      return true;
    }

    public override System.String ToString() {
      System.Text.StringBuilder b = new System.Text.StringBuilder();

      b.Append("Entry");
      b.Append("(");
      b.Append("thing=");
      b.Append(this.thing);
      b.Append(")");

      return b.ToString();
    }
  }
}
